### PR TITLE
Add twilio integration to send text messages

### PR DESCRIPTION
**Pull request by bertyhell**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +23,8 @@
   ],
   "dependencies": {
     "lodash": "~4.17.21",
-    "mandrill-api": "~1.0.45"
+    "mandrill-api": "~1.0.45",
+    "twilio": "^3.67.0"
```
Comment (horstenwillem) [inline package.json]: I prefer the use of ~ instead of ^

File: src/lib/email.service.ts [hunk 2]
```diff
@@ -16,5 +16,7 @@ export async function sendEmailWithTemplate(mailInfo: ITemplateRequest, mandrill
 
   let content: SendTemplateParams = mapTemplateEmail(mailInfo);
   content = _.merge(mandrillOptions, content);
+
+  console.log('test123456');
```
Comment (horstenwillem) [inline src/lib/email.service.ts]: Remove console log

File: src/lib/mailer.ts [hunk 3]
```diff
@@ -19,12 +19,15 @@ export function getClient(): Mandrill {
  */
 export async function sendTemplate(options: SendTemplateParams, client: Partial<Mandrill>): Promise<unknown> {
   return new Promise<unknown>((resolve, reject) => {
+    console.log('sendTemplate: ' + client.messages.sendTemplate);
```
Comment (horstenwillem) [inline src/lib/mailer.ts]: Remove console logs

File: src/lib/text-message.service.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import * as twilio from 'twilio';
+import { MessageInstance } from 'twilio/lib/rest/api/v2010/account/message';
+
+import { getTwilioAccountSid, getTwilioAuthToken } from '../config/client-config';
+
+/**
+ * Return a default MailClient (Mandrill in this case)
```
Comment (horstenwillem) [inline src/lib/text-message.service.ts]: Update JSDoc comment

File: src/lib/text-message.service.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  if (!accountSid) throw new Error('No twilio accountSid provided');
+  if (!authToken) throw new Error('No twilio authToken provided');
+  return twilio(accountSid, authToken);
+}
+
+/**
+ * Send text messages to phone numbers
+ * @returns {Object}
+ * @param fromNumber
+ * @param toNumber
```
Comment (horstenwillem) [inline src/lib/text-message.service.ts]: Add type to param documentation

File: tests/email.service.test.ts [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
 describe('email service', () => {
-  beforeAll(() => {
-    setMandrillApiKey('YY5pkBrTRAXKodN-0CL38g');
+  beforeEach(() => {
+    setMandrillApiKey('fake-mandril-api-key-for-testing');
   });
 
   afterEach(() => {
-    jest.clearAllMocks();
+    // jest.clearAllMocks();
```
Comment (horstenwillem) [inline tests/email.service.test.ts]: Re-enable clearAllMocks?

File: tests/email.service.test.ts [hunk 7]
```diff
@@ -24,6 +40,7 @@ describe('email service', () => {
         ],
       };
 
+      console.log('test123456');
```
Comment (horstenwillem) [inline tests/email.service.test.ts]: Remove console log

File: tsconfig.json [hunk 8]
```diff
@@ -15,7 +15,7 @@
     "noImplicitAny": true,
     "allowSyntheticDefaultImports": false,
     "removeComments": true,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (horstenwillem) [inline tsconfig.json]: Why disable the check for unused locals? 
What unused locals are in the project?

Comment (bertyhell) [reply tsconfig.json]: I was experimenting with a __mocks__ for the twilio client that had a constructor that had props that were not used.
This is no longer the case, so i'll re-enable it

**Review by horstenwillem — APPROVED** [review]

**Pull request merged**